common/src/node.rs

```diff
@@ -297,6 +297,42 @@ impl VoxelData {
             VoxelData::Solid(_) => true,
         }
     }
+
+    pub fn from_serializable(serializable: &SerializableVoxelData, dimension: u8) -> Option<Self> {
+        if serializable.voxels.len() != usize::from(dimension).pow(3) {
+            return None;
+        }
+
+        let mut data = vec![Material::Void; (usize::from(dimension) + 2).pow(3)];
+        let mut input_index = 0;
+        for x in 0..dimension {
+            for y in 0..dimension {
+                for z in 0..dimension {
+                    data[Coords([x, y, z]).to_index(dimension)] = serializable.voxels[input_index];
+                    input_index += 1;
+                }
+            }
+        }
+        Some(VoxelData::Dense(data.into_boxed_slice()))
+    }
+
+    pub fn to_serializable(&self, dimension: u8) -> SerializableVoxelData {
+        let VoxelData::Dense(data) = self else {
+            panic!("Only dense chunks can be serialized.");
+        };
+
+        let mut serializable: Vec<Material> = Vec::with_capacity(usize::from(dimension).pow(3));
+        for x in 0..dimension {
+            for y in 0..dimension {
+                for z in 0..dimension {
+                    serializable.push(data[Coords([x, y, z]).to_index(dimension)]);
+                }
+            }
+        }
```

Comment:
Reversing the nesting (z, y, x) is more conventional and will result in much more ordered memory access.

Reply:
Good catch. I definitely visualized this as having ordered memory access already, so this was just a mistake.